torchmetrics/functional/classification/hinge.py

```diff
@@ -0,0 +1,159 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, Tuple
+
+import torch
+from torch import Tensor, tensor
+
+from torchmetrics.utilities.data import to_onehot
+from torchmetrics.utilities.enums import DataType
+
+
+def _check_shape_and_type_consistency_hinge(
+        preds: Tensor,
+        target: Tensor,
+) -> DataType:
+    if target.ndim > 1:
+        raise ValueError(
+            f"The `target` should be one dimensional, got `target` with shape={target.shape}.",
+        )
+
+    if preds.ndim == 1:
+        if preds.shape != target.shape:
+            raise ValueError(
+                "The `preds` and `target` should have the same shape,",
+                f" got `preds` with shape={preds.shape} and `target` with shape={target.shape}.",
+            )
+        mode = DataType.BINARY
+    elif preds.ndim == 2:
+        if preds.shape[0] != target.shape[0]:
+            raise ValueError(
```

Comment:
seems we do not have tests for raising invalid values, mind add some

Reply:
Should be done already here (but codecov is still catching up haha): https://github.com/PyTorchLightning/metrics/blob/cfea3a077801bfb5e4dfa24d2c85a5a7d979d76c/tests/classification/test_hinge.py#L152